pkg/webhook/validator.go

```diff
@@ -288,14 +264,6 @@ func (v *Validator) validatePodSpec(ctx context.Context, namespace string, ps *c
 				continue
 			}
 			logging.FromContext(ctx).Errorf("ref: for %v", ref)
-			logging.FromContext(ctx).Errorf("container Keys: for %v", containerKeys)
-
-			if _, err := valid(ctx, ref, nil, containerKeys, ociremote.WithRemoteOptions(remote.WithAuthFromKeychain(kc))); err != nil {
```

Comment:
But if we do the above (add ability to configure what happens on 'no policy matches') as part of this PR, it might be cleaner to just do it now here, and we could use the 'FromContext' way to do this so that in this PR we configure this from the flag and later on we can change it to use ConfigMap if we so want to. Thoughts?

Reply:
Perhaps I am missing something here. I understand the new logic as the following:

`policies, err := config.ImagePolicyConfig.GetMatchingPolicies(ref.Name())`

If `policies == 0` and `err == nil` the function should admit the image.
If `policies > 0` and `err != nil` the function should deny the image.

I think this is what it is happening now.

Reply:
Yes, but I'm just thinking that some folks might want to start by only requiring policies for a subset of images. I just recall there being interest in away to specify deny everything that doesn't match a policy. A safety catch all. Does that make sense?
By having a flag, would allow us to specify:
 * If an image doesn't match any policy allow (behaviour with this PR)
 * If an image doesn't match any policy deny (with the additional flag)

Reply:
I'm not so convinced unless you're aware of folks asking for this flag. We have added many fields to handle all type of validations, `mode` and `action`, so users can decide how strict to be. This flag sounds like another additional mechanism to enable additional logic.
 
>  I just recall there being interest in away to specify deny everything that doesn't match a policy. A safety catch all.

Wouldn't it be easier to create a CIP with `glob: '**'` to any of my authorities ? With this CIP, I force all the images to satisfy a specific authority, and so deny all. 

Perhaps I am not seen the use case this flag will cover.

Reply:
Because the CIPs are 'and'ed together, if you add a CIP with a ** => deny all, everything would be denied.
So to your point that for policies we have added ability to configure what would happen when a policy is met, that's facts, and it's very flexible. But for the case where an image does not match a policy, there's no way to specify what should happen today. With the existing flag the behaviour is for better or for worse that it will be denied because it will fail without signatures (uses Fulcio if no secret given as you know) and if a secret is given it's roughly the equivalent to '**' -> key to validate, but with one big exception, this '**' -> key does not get 'AND'ed with the other policies.
So, I'm just worried that as it exists today, the behaviour will be a breaking change on how things behave from a functional perspective.
How about this?
 * Put this in, and let's create an issue where we can discuss this with the broader community.
 * Let's socialize the issue in the sigstore policy-controller and gather some thoughts on it.

Reply:
My bad!. I omitted the 'and'ed current behaviour when having multiple CIPs. I now understand the purpose of having a flag.

Okay, let's follow the steps that you described above.

Reply:
LOL, I'm glad you were able to decode what I was trying to say even with formatting screwing things up, yikes :)